### PR TITLE
Feat/be email newsletter subscription

### PR DESCRIPTION
**Pull request by Conradgabe**
[x] endpoint to subscribe and unsubscribe user
[x] function to send newsletter to subscribed user

### CODE REVIEW

File: backend/api/models/models.py [hunk 1]
```diff
@@ -205,6 +205,12 @@ class WatchlistItem(Base):
 
     company = relationship("Company")
 
+class NewsLetter(Base):
```
Comment (isoteriksoftware) [inline backend/api/models/models.py:208]: Use **NewletterSubscriber** as the model name

File: backend/api/models/models.py [hunk 2]
```diff
@@ -205,6 +205,12 @@ class WatchlistItem(Base):
...
+    __tablename__ = "newsletter"
+
+    id = Column("newletter_id", Integer, primary_key=True, index=True, autoincrement=True)
+    email = Column(String(64))
+    subscribed = Column(Boolean, default=False)
```
Comment (isoteriksoftware) [inline backend/api/models/models.py:213]: uhm, what is the use of the **subscribed** field?

File: backend/api/routes/newsletter.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+from api.crud.base import get_db
+from api.scripts.email import send_email
+from api.models.models import NewsLetter
+
+router = APIRouter()
+
+
+# subscribed user
+@router.post('/newsletter-subscribe', tags=['EmailSubscription'])
```
Comment (isoteriksoftware) [inline backend/api/routes/newsletter.py]: **newsletter-subscription** is a better name for this route

File: backend/api/routes/newsletter.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            users.subscribed = True
+            db.add(users)
+            db.flush()
+            db.commit()
+            db.refresh(users)
+            return {"isnewLetterSubscribed": {"email": users.email, "subscribed": users.subscribed}}
+        
+
+# cancel newletter subscription
+@router.post('/newsletter-unsubscribe', tags=['EmailSubscription'])
```
Comment (isoteriksoftware) [inline backend/api/routes/newsletter.py]: **newsletter-subscription** is also the better name here but now it should be a DELETE endpoint

**Review by isoteriksoftware — CHANGES_REQUESTED** [review]

File: backend/api/models/models.py (hunk 2 above)
Comment (Conradgabe) [reply backend/api/models/models.py:213]: To track the if the user is subscribed

File: backend/api/routes/newsletter.py (hunk 3 above)
Comment (Conradgabe) [reply backend/api/routes/newsletter.py]: Okay will change that

File: backend/api/models/models.py (hunk 2 above)
Comment (isoteriksoftware) [reply backend/api/models/models.py:213]: No bro, it shouldn't be there. Subscription verification should be independent of this model

**Review by isoteriksoftware — DISMISSED** [review]

**Review by isoteriksoftware — CHANGES_REQUESTED** [review]
Please make the appropriate changes

**Review by isoteriksoftware — APPROVED** [review]

**Pull request merged**